### ISSUE TITLE
Discordification of Nheko (feature request)

### ISSUE THREAD

**Issue by DonaldTsang** [mujx/nheko#462]
- [ ] Room feature support
    - [ ] Custom server-based and room-based emojis
       - Have custom server emojis downloaded and be usable as a basic feature
    - [ ] Emoji collection sharing
        - Similar to Telegram, Messenger, LINE and Whatsapp, but shareable and if possible, self-hosted or user-hosted
    - [ ] Emoji reaction for texts and files (useful for voting)
    - [ ] Automated (non-manual) key sharing
        - Some people are sick of needing to sharing key every time someone gets in the room, they would like to have auto key sharing once in an E2E room
    - [ ] Multi-room search
- [ ] Community support
    - [ ] Community id generation
    - [ ] Community room mass generation
    - [ ] Community room searching
- [ ] Role and access support
    - [ ] Mass room invitation by role or community
        - invitation to a server once application is approved
    - [ ] Mass room banning/kicking by role or community
        - for when "unfavorable" behavior by certain users happen
    - [ ] Community room sub-categorization and access
        - for visitor vs club member vs club staff separation
    - [ ] Role application and removal
        - for upgrading users to automatically bring them to certain rooms
    - [ ] Announcement rooms by role
        - rooms that allows viewing but not posting unless it is staff
    - [ ] Reaction allowance by role
       - for voting by certain role but not others
- [ ] audio chat support
    - [ ] self-host SIP room support (tying SIP with server)
    - [ ] self-host  Mumble room support (tying Mumble with server)
    - [ ] Radio/Playlist bot support
- [ ] collaborative support
    - [ ] self-host Etherpad/Firepad support (tying *pad with server)
    - [ ] self-host Hackmd support (tying Hackmd with server)

**Comment by mujx** [comment]
There are tons of features listed here. Please open specific issues for the rest of the features you would like to be supported.

- There is already an issue for audio chats.
- Widget-like features (etherpad etc) that Riot has, will not be implemented.
- Communities are not part of the spec and as such will not be supported anymore. They will be probably removed in the future.
- There is no spec for reactions, so they can't be implemented yet.

> Custom server-based and room-based emojis 

This is also spec related so nothing can be done until a spec is released.

**Comment by DonaldTsang** [comment]
@mujx Communities are in Synapse though, so it would be good to integrate that into the system
